flows/definition/flow.go

```diff
@@ -189,42 +187,34 @@ func (f *flow) ChangeLanguage(lang envs.Language) (flows.Flow, error) {
 
 	ll := copy.localization.(localization)
 
-	oldTranslation := make(languageTranslation) // current flow text extracted as a translation
-	newTranslation := ll[lang]
-
-	if newTranslation == nil {
-		return nil, errors.Errorf("no translation exists for %s", lang)
+	outTranslation := make(languageTranslation) // current flow text extracted out as a translation
+	inTranslation := ll[lang]                   // translation being imported in as new flow text
+	if inTranslation == nil {
+		inTranslation = languageTranslation{}
 	}
 
-	// all the locations where the incoming translation is missing text to replace current flow text
-	missing := make([]string, 0)
-
 	include := func(uuid uuids.UUID, property string, oldValues []string, w func([]string)) {
 		// save current flow text into a translation
 		if len(oldValues) > 0 {
-			oldTranslation.setTextArray(uuid, property, oldValues)
+			outTranslation.setTextArray(uuid, property, oldValues)
 		}
 
-		newValues := newTranslation.getTextArray(uuid, property)
+		newValues := inTranslation.getTextArray(uuid, property)
 
-		if len(oldValues) > 0 && len(newValues) == 0 {
-			missing = append(missing, fmt.Sprintf("%s/%s", uuid, property))
-		} else {
-			w(newValues) // update flow text in the definition
+		// if we have a translation, update flow text in the definition, if not then leave it as is
+		if len(newValues) > 0 {
```

Comment:
seems fine to let engine be dumb here and just default any missing translations to base text.. editor is the one tracking rules/categories translated and can make the decision about whether a flow is allowed to be translated